src/widgets/table.rs

```diff
@@ -1078,54 +1106,79 @@ impl<'a> StatefulWidget for Table<'a> {
         }
 
         // Draw rows
-        if self.rows.is_empty() {
-            return;
+        if !self.rows.is_empty() {
```

Comment:
Here, like in https://github.com/ratatui-org/ratatui/pull/516 I'd prefer to make the `render()` method call several smaller methods (`render_header()`, `render_rows()`, `render_footer()`) rather than increasing the size of this one method.